lr_train.py

```diff
@@ -0,0 +1,163 @@
+
+        return classifier.predict(test_array)  
+        
+    
+    def store(self, path):
+        '''Accepts a file path and stores the model at that file path using the pickle library.'''
+        
+        with open(path,'wb') as f:
+            pickle.dump(classifier, f)
+
+            
+def clean(text): 
+    '''Converts the input text into lower case, removes <br> tags, punctuation, whitespace, and stopwords. Lemmatizes the words.
+    Returns the processed words in a list.
+    
+        Args:
+        text(str): input text'''
+    
+    wnl = WordNetLemmatizer()
+    text = "".join([i.lower() for i in text.replace('<br', '') if i not in string.punctuation])
+    text = re.sub('\s+',' ', text)
+    text = [word for word in text.split() if word not in stopwords.words('english')]
+    text = ' '.join([wnl.lemmatize(word) for word in text])
+    
+    return text
+
+def compile_docs(data, col_name):
+    '''Cleans strings from the specified column from data, appends the cleaned strings to a list, returns the list of cleaned strings.
+        
+        Args:
+        data(df): a dataframe
+        col_name(str): name of text column from the dataframe
+    '''
+    
+    documents = []
+    for i in range(len(data)):
+        doc = data.loc[i, col_name]
+        processed_doc = clean(doc)
+        documents.append(processed_doc)
+        
+    return documents
+
+def tfidf_vectorize(docs):
+    '''Transforms text data into a matrix of TF-IDF features, using word as analyzer, and extracting both unigrams and bigrams.
+    Returns a document-term matrix.
+    
+        Args:
+        docs: a list (or another iterable) that generates strings'''
+    
+    vectorizer = TfidfVectorizer(analyzer='word', ngram_range=(1, 2))
+    tfidf_matrix = vectorizer.fit_transform(docs) # tfidf_matrix = csr.csr_matrix
+    
+    return tfidf_matrix
+
+def train_model(tfidf_arrays, labels):
+    '''Trains the model, returns the model.
+    
+        Args:
+        tfidf_arrays(array): arrays created by transforming text into a TF-IDF document-term matrix
+        labels(array): sentiment labels in the form of an array'''
+    
+    classifier = LogisticRegression().fit(tfidf_arrays, labels) 
+    
+    return classifier
+
+
+def main():
+    with ZipFile('imdb ratings.zip') as zf:
+        f = zf.open(PATH_TO_DATA_FILE)
+    
+    df = pd.read_csv(f)
+    smalldf = df[:DATA_LENGTH]
+    smalldf = smalldf.drop_duplicates(ignore_index = True)
+    smalldf.loc[smalldf[LABEL_COLUMN_NAME] == NEGATIVE_SENTIMENT_LABEL, LABEL_COLUMN_NAME] = NEGATIVE_SENTIMENT
+    smalldf.loc[smalldf[LABEL_COLUMN_NAME] == POSITIVE_SENTIMENT_LABEL, LABEL_COLUMN_NAME] = POSITIVE_SENTIMENT
+    
+    main.df = smalldf
+    
+    documents = compile_docs(smalldf, TEXT_COLUMN_NAME)
+        
+    X = tfidf_vectorize(documents).toarray()
+    y = smalldf[LABEL_COLUMN_NAME].values
+    
+    main.train_text, main.test_text, main.train_label, main.test_label = train_test_split(X, y)
```

Comment:
I'm surprised that this works!

I think this is somehow assigning members to the `main` global variable, which are then accessible elsewhere in the  module. We definitely don't want to do that I think.

Here is how I think the program should work at a high level:

1) We define a class for `LogisticRegressionClassifier`. It can be constructed with a trained `sklearn.LogisticRegression` classifier. That class exposes methods for `store`, `from_pickled`, `test`, and `predict`
2) We define some free (aka not attached to a class) functions for processing data and training a `sklearn.LogisticRegression` classifier
3) In `main`, we train a `sklearn.LogisticRegression` classifier using the free functions and the path to a dataset and then instantiate a `LogisticRegressionClassifier` object with that trained classifier
4) Then we call `test` with the test data and print out the results